### ISSUE TITLE
LunarGOO generates invalid GLSL for arrays in shader storage buffers

### ISSUE THREAD

**Issue by bfierz** [LunarG/LunarGLASS#13]
Hi,
I have tried to compile some of my shaders with LunarGOO to generate optimized shaders. However, the glsl generator seems to have problems with arrays in shader storage buffers. I have attached a test geometry shader, which generate the following faulty output:

```glsl
#version 430 core
// LunarGOO output
#extension GL_ARB_enhanced_layouts : enable
layout(points) in;
layout(triangle_strip) out;
layout(max_vertices = 3) out;
layout(std430) buffer VertexColours {
  vec4 Colour[];
} ;
layout(location=0) in VertexData {
  ivec3 Indices;
} In[1];
out VertexData {
  vec4 Colours;
} Out;

void main()
{
  Out.Colours = [In[0].Indices.x].Colour;
  EmitVertex();
  Out.Colours = [In[0].Indices.y].Colour;
  EmitVertex();
  Out.Colours = [In[0].Indices.z].Colour;
  EmitVertex();
  EndPrimitive();
}
```

[buffers_bug.geom.txt](https://github.com/LunarG/LunarGLASS/files/222598/buffers_bug.geom.txt)

**Comment by bfierz** [comment]
Seems to be a general issue with variable sized arrays at the end of a block. Removing the optimization passes in top to bottom translation results in the following error in BottomToGLSL:
"missing base expression for uniform makePointerExpression" in gla::GlslTarget::mapPointerExpression

**Comment by greg-lunarg** [comment]
FWIW, this works correctly on the LLVM3.6.2 branch. This branch uses LLVM 3.6.2 and has all fixes that master (based on LLVM 3.4) has.

**Comment by bfierz** [comment]
Thank you for the information. I can confirm that my example is now working. I'll stick to the 3.6.2 branch for the moment.
